pyexchange/leverj.py

```diff
@@ -375,39 +376,50 @@ def approve_token(self, token_address: str, amount: int) -> Transact:
     def deposit_ether(self, leverjobj: LeverjAPI, amount: Wad, gluon_block_number):
         assert(isinstance(leverjobj, LeverjAPI))
         assert(isinstance(amount, Wad))
-        assert(isinstance(gluon_block_number, None) or isinstance(gluon_block_number, int))
 
         custodian_account = self.address
         app_id = leverjobj.get_spot_exchange_id()
         if gluon_block_number is None:
-            gluon_block_number = leverjobj._http_authenticated("GET", "/api/v1", f"/plasma/{app_id}", None)['number'] +2
-            Transact(self, self.web3, self.abi, self.address, self._contract, "depositEther",[app_id], {'value': int(amount.value)}).transact()
-            return gluon_block_number
+            gluon_block_number = leverjobj._http_authenticated("GET", "/api/v1", f"/plasma/{app_id}", None)['number'] + 2
+            receipt = Transact(self, self.web3, self.abi, self.address, self._contract, "depositEther",[app_id], {'value': int(amount.value)}).transact()
+            return (gluon_block_number, receipt)
         else:
             current_gluon_block = leverjobj._http_authenticated("GET", "/api/v1", f"/plasma/{app_id}", None)['number']
             if (current_gluon_block < gluon_block_number):
-                return gluon_block_number
+                return (gluon_block_number, None)
             else:
-                return None
+                return (None, None)
 
     def deposit_token(self,  leverjobj: LeverjAPI, token_address: str, amount: int, gluon_block_number):
         assert(isinstance(leverjobj, LeverjAPI))
         assert(isinstance(token_address, str))
         assert(isinstance(amount, int))
-        assert(isinstance(gluon_block_number, None) or isinstance(gluon_block_number, int))
 
         custodian_account = self.address
         app_id = leverjobj.get_spot_exchange_id()
         if gluon_block_number is None:
-            gluon_block_number = leverjobj._http_authenticated("GET", "/api/v1", f"/plasma/{app_id}", None)['number'] +2
-            Transact(self, self.web3, self.abi, self.address, self._contract, "depositToken",[app_id, token_address, amount], {}).transact()
-            return gluon_block_number
+            gluon_block_number = leverjobj._http_authenticated("GET", "/api/v1", f"/plasma/{app_id}", None)['number'] + 2
+            receipt = Transact(self, self.web3, self.abi, self.address, self._contract, "depositToken",[app_id, token_address, amount], {}).transact()
```

Comment:
same as above, should we just return receipt?

Reply:
We need to return both, gluon_block_number tells us where we are in the process, the hash in the receipt is eventually sent back into post_pending_tx_hash to let let leverj know we just sent something and to look for that transaction hash in pending transactions on ethereum block chain.